packages/cozy-intent/src/api/models/environments.ts

```diff
@@ -2,12 +2,35 @@ import { Connection, EventsType, MethodsType } from 'post-me'
 
 import { NativeMethodsRegister } from '../../api'
 
+export interface WebviewSourceUri {
+  uri: string
+
+  baseUrl?: never
+  html?: never
```

Comment:
I don't understand that type. 

That is the first time I met the never, and this usage is not documented in the doc
https://www.tutorialsteacher.com/typescript/typescript-never

Reply:
`Never` means that those props can never have a value.

This notation allows to mutually exclude `WebviewSourceUri` and `WebviewSourceBaseUrl` when doing a union on those types.

You can set
```ts
const source: WebviewSource = { uri: 'https://foo.bar' }
```
or
```ts
const source: WebviewSource = { html: 'SOME_HTML', baseUrl: 'https://foo.bar' }
```
but the compiler will throw if you set

```ts
const source: WebviewSource = { uri: 'https://foo.bar', html: 'SOME_HTML', baseUrl: 'https://foo.bar' }
```
Because no type can have this combination (which is prevented by the `never` notation).

Reply:
ok i got it!
![Capture d’écran 2022-05-05 à 10 37 43](https://user-images.githubusercontent.com/8363334/166888904-b0dad859-1d0a-4a5d-b590-7c1c480faf70.png)